test: add test coverage - escalation boundary events migration

pedesen opened the pull request:
## Description

This PR adds E2E test coverage to ensure the migration works.
Mapping tasks inside sub processes with an escalation boundary event was already possible. 

The layout in the process definition files was updated due to readability reasons, which caused the big diff

Before:

![image](https://github.com/user-attachments/assets/0b9b43ba-5277-4c9c-bfa4-e74c48c25c75)

After:

![image](https://github.com/user-attachments/assets/08374e5b-35ca-4184-9e83-35bec4abe24c)


## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #24628 

Link to the test case: 
[User Can Migrate successfully a process with escalation boundary events](https://camunda.testrail.com/index.php?/cases/view/535967)

vitorwtavares commented on operate/client/e2e-playwright/tests/processInstanceMigration.spec.ts:
```diff
@@ -142,10 +145,10 @@ test.beforeAll(async ({request}) => {
 });
 
-test.describe.serial('Process Instance Migration', () => {
+test.describe.serial.only('Process Instance Migration', () => {
```
I believe this only is breaking the CI 😅

pedesen commented:
@mschoe Let me know if you have any questions. Here's a docker image of the branch you can use for testing: `registry.camunda.cloud/team-operate/camunda-operate:branch-fe-24628-migrate-escalation-boundary`.

ArezooAzar commented:
@pedesen : can you please review the test case [here](https://camunda.testrail.com/index.php?/cases/view/535967)

pedesen commented:
> @pedesen : can you please review the test case [here](https://camunda.testrail.com/index.php?/cases/view/535967)

@ArezooAzar I updated the instructions in the last step: added the zbctl command and described more precisely what's expected to happen. Looks good otherwise!
